S4TF_Tutorial_6.ipynb

```diff
@@ -0,0 +1,1356 @@
+        {
+          "output_type": "stream",
+          "text": [
+            "total training cat images: 1000\r\n",
+            "total training dog images: 1000\r\n",
+            "total validation cat images: 500\r\n",
+            "total validation dog images: 500\r\n",
+            "--\r\n",
+            "Total training images: 2000\r\n",
+            "Total validation images: 1000\r\n",
+            "<class 'list'>\r\n"
+          ],
+          "name": "stdout"
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "B1RxO8dzdF2g",
+        "colab_type": "text"
+      },
+      "source": [
+        "### Visualizing Training images"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "CFz4nx1LdGTZ",
+        "colab_type": "text"
+      },
+      "source": [
+        "* *We* can visualize our training images by creating functions to plot images through their paths or tensors, and then plotting a few of them."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "fCKdKItfmlep",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "func plotImages(_ image: Tensor<Float>) {\n",
+        "    let numpyImage = image.reshaped(to: [150, 150, 3]).makeNumpyArray()\n",
+        "    plt.imshow(numpyImage)\n",
+        "    plt.show()\n",
+        "}\n",
+        "\n",
+        "func plotImages(fromPath path: String) {\n",
+        "    let img = pil.Image.open(path)\n",
+        "    let image = np.array(img) * (1.0 / 255)\n",
+        "    plt.imshow(image)\n",
+        "    plt.show()\n",
+        "}"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "sZRtJODkfxgf",
+        "colab_type": "text"
+      },
+      "source": [
+        "# Data Augmentation"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "BpIp5hzAfx9P",
+        "colab_type": "text"
+      },
+      "source": [
+        "Overfitting often occurs when we have a small number of training examples. One way to fix this problem is to augment our dataset so that it has sufficient number and variety of training examples. Data augmentation takes the approach of generating more training data from existing training samples, by augmenting the samples through random transformations that yield believable-looking images. The goal is that at training time, your model will never see the exact same picture twice. This exposes the model to more aspects of the data, allowing it to generalize better.\n",
+        "\n",
+        "In S4TF we can implement this using the different Python Libraries through Python Interoperability. We can simply pass different transformations we would want to our dataset as a form of arguments and it will take care of applying it to the dataset during our training process."
```

Comment:
Always use "Swift for TensorFlow" instead of "S4TF" in any tutorials or official documentation.